File: app/src/main/java/andi/android/madegdk/adapter/MovieAdapter.java
package andi.android.madegdk.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.RatingBar;
import android.widget.TextView;

import com.bumptech.glide.Glide;

import java.util.ArrayList;

import andi.android.madegdk.R;
import andi.android.madegdk.model.Movie;

public class MovieAdapter extends BaseAdapter {

    private final Context context;
    private ArrayList<Movie> movies;

    public MovieAdapter(Context context) {
        this.context = context;
        movies = new ArrayList<>();
    }

    public void setMovies(ArrayList<Movie> movies) {
        this.movies = movies;
    }

    @Override
    public int getCount() {
        return movies.size();
    }

    @Override
    public Object getItem(int position) {
        return movies.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        if (convertView == null) {
            convertView = LayoutInflater.from(context).inflate(R.layout.item_movie, parent, false);
        }

        ViewHolder viewHolder = new ViewHolder(convertView);
        Movie movie = (Movie) getItem(position);
        viewHolder.bind(movie);
        return convertView;
    }

    private class ViewHolder {
        private TextView titleTV, dateTV;
        private ImageView posterIV;
        private RatingBar ratingBar;

        ViewHolder(View view) {
            titleTV = view.findViewById(R.id.titleTV);
            dateTV = view.findViewById(R.id.dateTV);
            posterIV = view.findViewById(R.id.posterIV);
            ratingBar = view.findViewById(R.id.ratingBar);
        }

        void bind(Movie movie) {
            titleTV.setText(movie.getTitle());
            dateTV.setText(movie.getDate());
            Glide.with(context)
                    .load(getDrawableId(movie.getPoster()))
                    .into(posterIV);
//            ratingBar.setMax(100);
            ratingBar.setRating(convertRatingToFloat(movie.getRating()));
        }
    }

    private Float convertRatingToFloat(Integer rating) {
        return rating * 5 / 100f;
    }

    private Integer getDrawableId(String drawableName) {
        return context.getResources().getIdentifier(drawableName, "drawable", context.getPackageName());
    }
}

File: app/src/main/java/andi/android/madegdk/ui/home/movie/adapter/MovieAdapter.kt
package andi.android.madegdk.ui.home.movie.adapter

import andi.android.madegdk.BuildConfig
import andi.android.madegdk.R
import andi.android.madegdk.database.FavoriteMovieHelper
import andi.android.madegdk.model.Movie
import andi.android.madegdk.utils.normalizeRating
import android.content.Context
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.animation.AnimationUtils
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import kotlinx.android.synthetic.main.item_movie.view.*


class MovieAdapter(private val context: Context?, private val listener: (Movie, Boolean, Int) -> Unit) :
        RecyclerView.Adapter<MovieViewHolder>() {

    private val mData = ArrayList<Movie>()

    override fun onCreateViewHolder(p0: ViewGroup, p1: Int): MovieViewHolder {
        return MovieViewHolder(
                context,
                LayoutInflater.from(p0.context).inflate(
                        R.layout.item_movie,
                        p0,
                        false
                )
        )
    }

    override fun getItemCount(): Int = mData.size

    override fun onBindViewHolder(p0: MovieViewHolder, p1: Int) {
        if (context != null) {
            p0.bindItem(context, mData[p1], listener)
        }
    }

    fun setMovies(movies: ArrayList<Movie>) {
        mData.clear()
        mData.addAll(movies)
        notifyDataSetChanged()
    }

    fun addMovies(movies: ArrayList<Movie>) {
        mData.addAll(movies)
        notifyDataSetChanged()
    }

    fun updateItem(position: Int) {
        notifyItemChanged(position)
    }

}

class MovieViewHolder(context: Context?, view: View) : RecyclerView.ViewHolder(view) {

    private val favoriteMovieHelper = FavoriteMovieHelper.getInstance(context)

    fun bindItem(context: Context, movie: Movie, listener: (Movie, Boolean, Int) -> Unit) {
        itemView.itemParentCV.animation = AnimationUtils.loadAnimation(context, R.anim.item_animation_slide_from_left)
        itemView.titleTV.text = movie.title
        itemView.dateTV.text = movie.date
        Glide.with(context)
                .load("${BuildConfig.IMAGE_URL}t/p/w185${movie.poster}")
                .into(itemView.posterIV)
        itemView.ratingBar.rating = normalizeRating(movie.rating)

        itemView.setOnClickListener {
            listener(movie, itemView.favoriteBT.isChecked, layoutPosition)
        }
        itemView.favoriteBT.setOnClickListener {
            if (itemView.favoriteBT.isChecked) {
                val aMovie = Movie(movie.movieId, movie.poster, movie.backdrop, movie.title, movie.date, movie.rating, movie.overview)
                favoriteMovieHelper?.insertMovieFavorite(aMovie)
                Log.d("FAV", favoriteMovieHelper?.getAllFavoriteMovies().toString())
            } else {
                favoriteMovieHelper?.deleteMovieFavorite(movie.movieId)
                Log.d("FAV", favoriteMovieHelper?.getAllFavoriteMovies().toString())
            }
        }
        if (favoriteMovieHelper != null) {
            itemView.favoriteBT.isChecked = favoriteMovieHelper.isFavorite(movie.movieId)
        }
    }
}